### PR TITLE
docs: reaction platform install docs

### LINKED ISSUE #676: Add Reaction Platform install instructions

### PR DESCRIPTION
**Pull request by machikoyasuda**
Resolves #676 
Impact: **critical**  
Type: **docs**

## Issue

- We need a canonical place for installation instructions.

## Solution & Screenshots

- We will have all the official instructions here in docs.
- We will then update the READMEs for `reaction`, `reaction-next-starterkit`, `reaction-platform` to link to these installation instructions.

## Breaking changes
- We need to deprecate or remove the old OSX, Linux, Windows instructions

## Post-PR actions
- Update `reaction`, `reaction-next-starterkit`, `reaction-platform`  installation instruction in READMEs to link to here.

## Testing & Reviewing
1. Need someone to test these instructions on a fresh install with OSX (@spencern), Linux (@focusaurus or @rosshadden) and Windows (@jeffcorpuz).
2. Need someone to test that these applications names are in line with our branding and architecture (@ticean)

### CODE REVIEW

File: public-docs/installation-reaction-platform.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Follow these instructions to run the entire Reaction Platform. 
+
+The Platform will install and run the the entire suite of Reaction services in the these directories:
+
+| Directory                                                                                  | Services                        |
+| ------------------------------------------------------------------------------------------ | ------------------------------- |
+| [`reaction`](https://github.com/reactioncommerce/reaction)                                 | GraphQL API, Classic UI, Mongo  |
+| [`reaction-hydra`](https://github.com/reactioncommerce/reaction-hydra)                     | Authentication server, Postgres |
+| [`reaction-next-starterkit`](https://github.com/reactioncommerce/reaction-next-starterkit) | Next.js storefront application  |
```
Comment (machikoyasuda) [inline public-docs/installation-reaction-platform.md]: @ticean - Does the way I broke down the apps by directory, and then listed each service, look good to you?

And as for naming, how is `GraphQL API, Classic UI, Mongo`? Or would you prefer: `Reaction API, Mongo, Classic UI` or something else?

File: public-docs/installation-reaction-platform.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+## Installation
+
+1. Requirements:
+
+- Install Docker and Docker Compose ([Mac](https://docs.docker.com/docker-for-mac/install/) | [Windows](https://docs.docker.com/docker-for-windows/install/)* | [Linux](https://docs.docker.com/compose/install/#install-compose)**) 
+- Install [Node.js](https://nodejs.org/en/)
+- A GitHub account with a configured SSH key
+
+> *Windows: Reaction Platform has not been fully tested in Windows at this time.
```
Comment (machikoyasuda) [inline public-docs/installation-reaction-platform.md]: @jeffcorpuz Can you clarify what you think we notes we should add for Windows users on line 24?

File: public-docs/installation-reaction-platform.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+> **Linux: Docker Compose is included when installing Docker on Mac and Windows, but will need to be installed separately on Linux.
```
Comment (machikoyasuda) [inline public-docs/installation-reaction-platform.md]: @rosshadden @focusaurus Does this sound good for Linux users? Are there other things Linux users need to know before they go ahead with this step?

File: public-docs/installation-reaction-platform.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+4. Now you're ready to install. Run this command to bootstrap and start all of the services:
+
+```sh
+make
+```
+
+This process may take some time. The Platform is checking that dependencies are present, cloning the sub-projects from GitHub, downloading and building Docker images, and starting services.
+
+5. As the various repositories are downloading and services are starting, check the logs by running `docker-compose logs -f` in the specific repository. For example, to display the `reaction` logs, run:
```
Comment (machikoyasuda) [inline public-docs/installation-reaction-platform.md]: @ticean Wasn't sure what the best step was here. I want something to include a step to help users understand that there is something going on behind the scenes, and give them ways to figure out what is going on behind the scenes. I was also thinking of showing what the logs should look like when they know they are ready.. like:

---
5. Once you see that the services started, you can start viewing the logs for each service:

```
Starting reaction-hydra_postgres_1 ... done
Starting reaction-hydra_hydra-migrate_1 ... done
Starting reaction-hydra_hydra_1         ... done
reaction_mongo_1 is up-to-date
Starting reaction_mongo-init-replica_1 ... done
Starting reaction_reaction_1           ... done
Starting reaction-next-starterkit_web_1 ... done
```

File: public-docs/installation-reaction-platform.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+## Installation
+
+1. Requirements:
+
+- Install Docker and Docker Compose ([Mac](https://docs.docker.com/docker-for-mac/install/) | [Windows](https://docs.docker.com/docker-for-windows/install/)<sup>*</sup> | [Linux](https://docs.docker.com/compose/install/#install-compose)<sup>**</sup>) 
+- Install [Node.js](https://nodejs.org/en/)
+- A GitHub account with a configured SSH key
+
+> <sup>*</sup>Windows: Reaction Platform has not been fully tested in Windows at this time.
```
Comment (focusaurus) [inline public-docs/installation-reaction-platform.md]: Minor edit: `has not been fully tested on Windows at this time.`

File: public-docs/installation-reaction-platform.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+Learn more about [Reaction Platform](https://github.com/reactioncommerce/reaction-platform).
+
+### Docker commands
+
+Run these commands from each respective application's directory:
+
+| Command                  | Description                         |
+| ------------------------ | ----------------------------------- |
+| `docker-compose logs -f` | Display logs and follow log output. |
```
Comment (focusaurus) [inline public-docs/installation-reaction-platform.md]: Looks like this is still maybe WIP? We should probably document start, stop, and destroy commands.

File: public-docs/installation-reaction-platform.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+To display the `reaction-next-starterkit` logs, run:
+
+```sh
+cd reaction-next-starterkit
+docker-compose logs -f
+```
+
+6. Once the initial `make`  process finishes, the services will be accessible at the following URLs:
```
Comment (focusaurus) [inline public-docs/installation-reaction-platform.md]: Probably need to add a note here about expected wait time. Immediately after `make` finishes, there's still a long wait before the web apps are running.

File: public-docs/installation-reaction-platform.md (hunk 6 above)
Comment (machikoyasuda) [reply public-docs/installation-reaction-platform.md]: We have a lotta Docker documented here already:  ```Learn more about [developing in Docker](https://docs.reactioncommerce.com/docs/next/installation-docker-development#run-the-apps) and [troubleshooting Docker development](https://docs.reactioncommerce.com/docs/next/troubleshooting-development#docker-issues). ```

This part is more like a very brief overview to get started, and the TLDR is linked below on line 101.

**Pull request merged**